custom_components/virtual_keys/__init__.py

```diff
@@ -114,7 +114,7 @@ async def delete_token(
     for user in await hass.auth.async_get_users():
         for token in list(user.refresh_tokens.values()):
             if (token.id == msg.get("token_id")):
-                await hass.auth.async_remove_refresh_token(token)
+                hass.auth.async_remove_refresh_token(token)
```

Comment:
Thanks for contributing!

can you please explain why we need to remove the async here? the method is an async method.

Reply:
The await generate an error in front and futhermore the the return of this function is None so we don't used the result. So I remove it to avoid the error.

Reply:
Sorry, I see that the link is broken...
https://github.com/home-assistant/core/blob/8cae8edc5557828f97dd2f9938c3bafdda49d21b/homeassistant/auth/__init__.py#L530

Reply:
Investigated a bit on my side and you are right. The error yields because `async_remove_refresh_token` is not an async method, despite the name.
@Darkdragon14 can you please create a new PR with this commit only?

Reply:
for the date picker, added  [here](https://github.com/kcsoft/virtual-keys/pull/10), please see if this meets your needs

Reply:
Yes, I will do this soon 
It's good for your implementation ! 😄